src/local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.cc

```diff
@@ -0,0 +1,228 @@
+    _env->ReleaseByteArrayElements(_bytes, reinterpret_cast<jbyte *>(PID), 0);
+  }
+};
+
+/*
+ * Class:     org_ray_spi_impl_DefaultLocalSchedulerClient
+ * Method:    _init
+ * Signature: (Ljava/lang/String;[B[BZJ)J
+ */
+JNIEXPORT jlong JNICALL
+Java_org_ray_spi_impl_DefaultLocalSchedulerClient__1init(JNIEnv *env,
+                                                         jclass,
+                                                         jstring sockName,
+                                                         jbyteArray wid,
+                                                         jbyteArray actorId,
+                                                         jboolean isWorker,
+                                                         jlong numGpus) {
+  // 	native private static long _init(String localSchedulerSocket,
+  //     byte[] workerId, byte[] actorId, boolean isWorker, long numGpus);
+  UniqueIdFromJByteArray w(env, wid), a(env, actorId);
+  const char *nativeString = env->GetStringUTFChars(sockName, JNI_FALSE);
+  auto client = LocalSchedulerConnection_init(nativeString, *w.PID, isWorker);
+  env->ReleaseStringUTFChars(sockName, nativeString);
+  return (jlong) client;
+}
+
+/*
+ * Class:     org_ray_spi_impl_DefaultLocalSchedulerClient
+ * Method:    _submitTask
+ * Signature: (JLjava/nio/ByteBuffer;II)V
+ */
+JNIEXPORT void JNICALL
+Java_org_ray_spi_impl_DefaultLocalSchedulerClient__1submitTask(
+    JNIEnv *env,
+    jclass,
+    jlong c,
+    jbyteArray cursorId,
+    jobject buff,
+    jint pos,
+    jint sz) {
+  // task -> TaskInfo (with FlatBuffer)
+  // native private static void _submitTask(long client, /*Direct*/ByteBuffer
+  // task);
+  auto client = reinterpret_cast<LocalSchedulerConnection *>(c);
+  TaskSpec *task =
+      reinterpret_cast<char *>(env->GetDirectBufferAddress(buff)) + pos;
+  std::vector<ObjectID> execution_dependencies;
+  if (cursorId != NULL) {
+    UniqueIdFromJByteArray cursor_id(env, cursorId);
+    execution_dependencies.push_back(*cursor_id.PID);
+  }
+  TaskExecutionSpec taskExecutionSpec =
+      TaskExecutionSpec(execution_dependencies, task, sz);
+  local_scheduler_submit(client, taskExecutionSpec);
+}
+
+/*
+ * Class:     org_ray_spi_impl_DefaultLocalSchedulerClient
+ * Method:    _getTaskTodo
+ * Signature: (J)[B
+ */
+JNIEXPORT jbyteArray JNICALL
+Java_org_ray_spi_impl_DefaultLocalSchedulerClient__1getTaskTodo(JNIEnv *env,
+                                                                jclass,
+                                                                jlong c) {
+  // native private static ByteBuffer _getTaskTodo(long client);
+  auto client = reinterpret_cast<LocalSchedulerConnection *>(c);
+  int64_t task_size = 0;
+
+  // TODO: handle actor failure later
+  TaskSpec *spec = local_scheduler_get_task(client, &task_size);
+
+  jbyteArray result;
+  result = env->NewByteArray(task_size);
+  if (result == NULL) {
+    return NULL; /* out of memory error thrown */
```

Comment:
Is the plan to let the caller check for this and throw a Java exception?

Reply:
yes, currently this situation will throw a Exception in java